server/service/src/sync/sync_on_central/mod.rs

```diff
@@ -126,13 +152,75 @@ pub async fn push(
         repo.upsert_one(&buffer_row)?;
     }
 
-    // TODO we need to trigger integrate records for just 1 site?
-    // See issue: https://github.com/msupply-foundation/open-msupply/issues/3294
-    if total_records <= records_in_this_batch {
-        service_provider.sync_trigger.trigger();
+    if is_last_batch {
+        spawn_integration(service_provider.clone(), response.site_id);
     }
 
     Ok(SyncPushSuccessV6 {
         records_pushed: records_in_this_batch,
     })
 }
+
+pub async fn get_site_status(
+    SiteStatusRequestV6 { sync_v5_settings }: SiteStatusRequestV6,
+) -> Result<SiteStatusV6, SyncParsedErrorV6> {
+    use SyncParsedErrorV6 as Error;
+
+    if !CentralServerConfig::is_central_server() {
+        return Err(Error::NotACentralServer);
+    }
+
+    let response = SyncApiV5::new(sync_v5_settings)
+        .map_err(|e| Error::OtherServerError(format_error(&e)))?
+        .get_site_info()
+        .await
+        .map_err(Error::from)?;
+
+    let code = match is_integrating(response.site_id) {
+        true => SiteStatusCodeV6::IntegrationInProgress,
+        false => SiteStatusCodeV6::Idle,
+    };
+
+    Ok(SiteStatusV6 { code })
+}
+
+fn is_integrating(site_id: i32) -> bool {
+    let sites_being_integrated = SITES_BEING_INTEGRATED.read().unwrap();
+    sites_being_integrated.contains(&site_id)
+}
+
+fn set_integrating(site_id: i32, is_integrating: bool) {
+    let mut sites_being_integrated = SITES_BEING_INTEGRATED.write().unwrap();
+
+    if is_integrating {
+        sites_being_integrated.push(site_id);
+    } else {
+        sites_being_integrated.retain(|id| *id != site_id);
+    }
+}
+
+fn spawn_integration(service_provider: Arc<ServiceProvider>, site_id: i32) -> () {
+    tokio::spawn(async move {
+        let ctx = match service_provider.basic_context() {
```

Comment:
The compiler is sad here once I start doing anything with `StorageConnection`, as the `transaction_level` doesn't implement `Sync`... had a bit of a play but feels like I'm shooting in the dark a bit now - send help?